From e984b5cfbb09263a5655162b27649f643550a582 Mon Sep 17 00:00:00 2001
From: Mehdi Chinoune <mehdi.chinoune@hotmail.com>
Date: Mon, 14 Mar 2022 12:37:20 +0100
Subject: [PATCH 1/2] Fix detecting MINGW_ARCH

${MINGW_PATH}/${MINGW_CHOST} path doesn't exist on CLANG* environments
---
 CMakeScripts/ConfigEnvMinGW.cmake | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeScripts/ConfigEnvMinGW.cmake b/CMakeScripts/ConfigEnvMinGW.cmake
index 4c0ff6facd..8233f0fadd 100644
--- a/CMakeScripts/ConfigEnvMinGW.cmake
+++ b/CMakeScripts/ConfigEnvMinGW.cmake
@@ -9,7 +9,7 @@
 # -----------------------------------------------------------------------------
 message(STATUS "Configuring MinGW environment:")
 
-if("$ENV{MSYSTEM_CHOST}" STREQUAL "")
+if("$ENV{MINGW_CHOST}" STREQUAL "")
   message(WARNING "  Could not detect MinGW build environment. We recommend building with MSYS2. Proceed at your own risk!")
 else()
   message(STATUS "  Detected MinGW environment provided by MSYS2")
@@ -21,10 +21,10 @@ endif()
 # -----------------------------------------------------------------------------
 
 # Try to determine the MinGW processor architecture.
-if(EXISTS "${MINGW_PATH}/i686-w64-mingw32")
+if("$ENV{MINGW_CHOST}" STREQUAL "i686-w64-mingw32")
   set(HAVE_MINGW64 OFF)
   set(MINGW_ARCH i686-w64-mingw32)
-elseif(EXISTS "${MINGW_PATH}/x86_64-w64-mingw32")
+elseif("$ENV{MINGW_CHOST}" STREQUAL "x86_64-w64-mingw32")
   set(HAVE_MINGW64 ON)
   set(MINGW_ARCH x86_64-w64-mingw32)
 else()
-- 
GitLab


From dd273c41155ad392b802fdd68f58564ef94de03a Mon Sep 17 00:00:00 2001
From: Mehdi Chinoune <mehdi.chinoune@hotmail.com>
Date: Mon, 14 Mar 2022 12:39:24 +0100
Subject: [PATCH 2/2] Remove MINGW_ARCH_PATH

MSYS2 no longer install libraries there.
---
 CMakeLists.txt                    |  1 -
 CMakeScripts/ConfigEnvMinGW.cmake | 12 ------------
 2 files changed, 13 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 34830fd4e7..7e0b64ba7a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -298,7 +298,6 @@ message("")
 message("HAVE_MINGW64:            ${HAVE_MINGW64}")
 message("MINGW_PATH:              ${MINGW_PATH}")
 message("MINGW_ARCH:              ${MINGW_ARCH}")
-message("MINGW_ARCH_PATH:         ${MINGW_ARCH_PATH}")
 endif()
 
 message("------------------------------------------------------------------------")
diff --git a/CMakeScripts/ConfigEnvMinGW.cmake b/CMakeScripts/ConfigEnvMinGW.cmake
index 8233f0fadd..222582da35 100644
--- a/CMakeScripts/ConfigEnvMinGW.cmake
+++ b/CMakeScripts/ConfigEnvMinGW.cmake
@@ -31,9 +31,6 @@ else()
   message(FATAL_ERROR "Unable to determine MinGW processor architecture. Are you using an unsupported MinGW version?")
 endif()
 
-# Path to processor architecture specific binaries and libs.
-set(MINGW_ARCH_PATH "${MINGW_PATH}/${MINGW_ARCH}")
-
 set(MINGW_BIN "${MINGW_PATH}/bin")
 
 if(NOT EXISTS ${MINGW_BIN})
@@ -61,15 +58,6 @@ endif()
 # Add MinGW headers to compiler include path.
 include_directories(SYSTEM ${MINGW_INCLUDE})
 
-set(MINGW_ARCH_LIB "${MINGW_ARCH_PATH}/lib")
-
-if(NOT EXISTS ${MINGW_ARCH_LIB})
-  message(FATAL_ERROR "MinGW-w64 toolchain libraries directory does not exist: ${MINGW_ARCH_LIB}")
-endif()
-
-# Add MinGW toolchain libraries to linker path.
-link_directories(${MINGW_ARCH_LIB})
-
 # -----------------------------------------------------------------------------
 # MSYS CHECKS
 # -----------------------------------------------------------------------------
-- 
GitLab

